src/net.cpp

```diff
@@ -348,6 +348,11 @@ CNode* CConnman::FindNode(const CService& addr)
     return nullptr;
 }
 
+bool CConnman::AlreadyConnectedToPeer(const CAddress& addr)
+{
+    return FindNode(static_cast<CNetAddr>(addr)) || FindNode(addr.ToStringIPPort());
```

Comment:
It might be worth adding a comment about why we search both for the `CNetAddr` _and_ the ip/port string. It was added here: https://github.com/bitcoin/bitcoin/commit/9bab521df895c149579b9e64931405c56b008afb#diff-00021eed586a482abdb09d6cdada1d90115abe988a91421851960e26658bed02R1416 in the PR to add support for SOCKS5.

Reply:
Care to suggest a comment?  I'm not sure my understanding of when we might be looking up by `CNetAddr` versus string is precise enough to say the right thing here.

Reply:
I was hoping you might have a better understanding of this than me :grin: . My understanding of how hostnames/proxies work in Bitcoin Core is certainly too hazy to suggest any wording here.

I'm going to mark this as resolved. It'd be good to document why we're doing this, but this PR isn't changing the behavior, so it's certainly not a blocker.

Reply:
@ariard If there is a name proxy, we never resolve ourselves, and instead make a proxy connection to the specified name (by asking the proxy to do the resolving for us). In that case there is no `CNetAddr` known.

Reply:
@sipa But `CAddress& addr` is a `CNetAddr` and `addr.ToStringIPPort()` generates a string from the `CNetAddr` member variables? As far as I can see there's no way to initialise `CNetAddr::m_addr` to something like `::1` or `0:0:0:0:0:0:0:1` which would be disambiguated by converting to a canonical string?

Reply:
Oh, right. What I said isn't relevant here.